### PR TITLE
Integration tests for metrics in the Placement Service and a few fixes

### PR DESCRIPTION
**Pull request by elena-kolevska**
# Description

Adding integration tests before the Open Census to the Open Telemetry change, so we can confirm that no regressions have been introduced.

It also addresses https://github.com/dapr/dapr/issues/8175

## Issue reference

https://github.com/dapr/dapr/issues/7225

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [X] Code compiles correctly
- [X] Created/updated tests
- [x] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Specification has been updated / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Provided sample for the feature / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/integration/framework/process/metrics/metrics.go [hunk 1]
```diff
@@ -65,6 +74,35 @@ func New(t *testing.T, ctx context.Context, url string) *Metrics {
...
 }
 
-func (m *Metrics) GetMetrics() map[string]float64 {
+func (m *Metrics) All() map[string]float64 {
 	return m.metrics
 }
+
+// MatchMetric returns all metrics that contain all the substrings in the key
+// This is useful because of the way we serialize labels in the metrics name
+func (m *Metrics) MatchMetric(args ...string) []Metric {
```
Comment (JoshVanL) [inline tests/integration/framework/process/metrics/metrics.go:83]:
```suggestion
func (m *Metrics) MatchMetric(substrings ...string) []Metric {
```

File: tests/integration/framework/process/metrics/metrics_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package metrics
```
Comment (JoshVanL) [inline tests/integration/framework/process/metrics/metrics_test.go:1]: Nit: copyright

**Review by JoshVanL — CHANGES_REQUESTED** [review]
Big fan 😄

File: tests/integration/framework/process/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	result := make([]Metric, 0)
+
+	// Iterate over all key-value pairs in the map
+	for key, value := range m.metrics {
+		matchesAll := true
+
+		// Check if all args are present in the key
+		for _, arg := range args {
+			if !strings.Contains(key, arg) {
```
Comment (elena-kolevska) [inline tests/integration/framework/process/metrics/metrics.go:92]:
```suggestion
		// Check if all substrings are present in the key
		for _, substring := range substrings {
			if !strings.Contains(key, substring) {
```

**Review by JoshVanL — DISMISSED** [review]

**Review by yaron2 — APPROVED** [review]

**Review by JoshVanL — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**